test/hypothesis_utils.py

```diff
@@ -107,46 +136,92 @@ def array_shapes(draw, min_dims=1, max_dims=None, min_side=1, max_side=None):
         st.integers(min_side, max_side), min_size=min_dims, max_size=max_dims
     ).map(tuple))
 
-"""Strategy for generating test cases for quantized tensors.
+
+"""Strategy for generating test cases for tensors.
 The resulting tensor is in float32 format.
 
 Args:
-    min_batch, max_batch: Range to generate `nbatch`
-    min_in_channels, max_in_channels: Range to generate `iChannels`
-    min_out_channels, max_out_channels: Range to generate `oChannels`
+    shapes: Shapes under test for the tensor. Could be either a hypothesis
+            strategy, or an iterable of different shapes to sample from.
+    elements: Elements to generate from for the returned data type.
+              If None, the strategy resolves to float within range [-1e6, 1e6].
+    qparams: Instance of the qparams strategy. This is used to filter the tensor
+             such that the overflow would not happen.
+
+Generates:
+    X: Tensor of type float32. Note that NaN and +/-inf is not included.
+    qparams: (If `qparams` arg is set) Quantization parameters for X.
+        The returned parameters are `(scale, zero_point, quantization_type)`.
+        (If `qparams` arg is None), returns None.
+"""
+@st.composite
+def tensor(draw, shapes=None, elements=None, qparams=None):
+    if isinstance(shapes, SearchStrategy):
+        _shape = draw(shapes)
+    else:
+        _shape = draw(st.sampled_from(shapes))
+    if qparams is None:
+        if elements is None:
+            elements = st.floats(-1e6, 1e6)
```

Comment:
should this be float_min and float_max?

Reply:
float_min and float_max are replaced with `elements`